### ISSUE TITLE
`-Wl,-mllvm` causes `unsupported linker arg: -mllvm`

### ISSUE THREAD

**Issue by mrexodia** [ziglang/zig#22291]
### Zig Version

0.14.0-dev.2546+0ff0bdb4a

### Steps to Reproduce and Observed Behavior

Compile simple `main.cpp`:

```sh
zig c++ -target x86_64-linux-gnu main.cpp -o main -flto -fuse-ld=lld -v -Wl,-mllvm,-lto-embed-bitcode=optimized
```

### Expected Behavior

Zig should include `-mllvm -lto-embed-bitcode=optimized` in the `ld.lld` command line.

Using the following command line on Linux works fine:

```sh
clang++ main.cpp -o main -flto -fuse-ld=lld -v -Wl,-mllvm,-lto-embed-bitcode=optimized
```

An alternative command line (also rejected) is: `-Wl,--plugin-opt=-lto-embed-bitcode=optimized`

This option is used to add a `.llvmbc` section to the final executable, which contains all of the bitcode, but there are other options as well. Related issue: #11204, related pull request https://github.com/llvm/llvm-project/pull/79390

**Comment by alexrp** [comment]
Note that we're planning on dropping the LLD dependency Real Soon Now, so this option would only be supportable when using LLD through the Zig build system, and with Zig's LLVM backend. Usage of Zig's self-hosted linker and/or self-hosted backend(s) is fundamentally incompatible with this.

(Does Apple still require apps to ship with embedded bitcode?)

**Comment by mrexodia** [comment]
Yeah this is for usage with `zig c++` only. I do not need to embed the bitcode for Apple, it's to do my own post-processing on bitcode of the whole binary and zig is the most convenient cross compiler to play around with things!
